app/backend/src/clj/teet/db_api/db_api_handlers.clj

```diff
@@ -75,51 +76,64 @@
 (defn- request [handler-fn]
   (not-modified/wrap-not-modified
    (fn [req]
-     (try
-       (let [user (some->> req jwt-token (jwt-token/verify-token
-                                          (environment/config-value :auth :jwt-secret)))]
-         (log/with-context
-           {:user (str (:user/id user))}
-           (let [conn (environment/datomic-connection)
-                 db (d/db conn)
-                 aconn (delay (environment/asset-connection))
-                 ctx {:conn conn
-                      :db db
-                      :asset-conn aconn
-                      :asset-db (delay (d/db @aconn))
-                      :user (merge user
-                                   (when-let [user-id (:user/id user)]
-                                     (user-db/user-info db [:user/id user-id])))
-                      :session (:session req)
-                      :headers (:headers req)}
-                 request-payload (transit/transit-request req)
-                 response (handler-fn ctx request-payload)]
-             (case (:format (meta response))
-               ;; If :format meta key is :raw, send output as ring response
-               :raw response
+     (let [cleanup (atom [])]
+       (try
+         (let [user (some->> req jwt-token (jwt-token/verify-token
+                                            (environment/config-value :auth :jwt-secret)))]
+           (log/with-context
+             {:user (str (:user/id user))}
+             (let [conn (environment/datomic-connection)
+                   db (d/db conn)
+                   aconn (delay (environment/asset-connection))
+                   pg (delay
+                        (let [c (environment/get-pg-connection)]
+                          (swap! cleanup conj
+                                 #(do
+                                    (log/debug "Closing PostgreSQL connection" c)
+                                    (.close c)))
+                          {:connection c}))
+                   ctx {:conn conn
+                        :db db
+                        :pg pg
```

Comment:
we add a `:pg` key to context, which if used will get a connection from the pool and register cleanup function that is run below in the finally block.